src/sentry/api/authentication.py

```diff
@@ -127,9 +132,59 @@ def relay_from_id(request, relay_id) -> Tuple[Optional[Relay], bool]:
 
 
 class QuietBasicAuthentication(BasicAuthentication):
+    _hybrid_cloud_rollout_level = -1
+
     def authenticate_header(self, request: Request) -> str:
         return 'xBasic realm="%s"' % self.www_authenticate_realm
 
+    def _use_authenticated_token(self) -> bool:
+        return (
+            SiloMode.get_current_mode() != SiloMode.MONOLITH
+            or options.get("hybrid_cloud.authentication.use_authenticated_token")
+            >= self._hybrid_cloud_rollout_level
+            or in_test_environment()
+        )
+
+    def _use_rpc_user(self) -> bool:
+        return (
+            SiloMode.get_current_mode() != SiloMode.MONOLITH
+            or options.get("hybrid_cloud.authentication.use_rpc_user")
+            >= self._hybrid_cloud_rollout_level
+            or in_test_environment()
+        )
+
+    def transform_auth(
+        self,
+        user: int | User | RpcUser | None | AnonymousUser,
+        auth: Any,
```

Comment:
We overload `auth` often, could we use a name like `auth_token` 🤷

Reply:
Yes -- but, `auth` is the canonical word for the the thing we're expecting here (whatever went into `request.auth`.  What if I call it `request_auth`?

Reply:
request_auth works for me.